src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -162,17 +164,108 @@ MdnsContexts MdnsContexts::sInstance;
 
 namespace {
 
+static void OnRegisterRecord(DNSServiceRef sdRef, DNSRecordRef recordRef, DNSServiceFlags flags, DNSServiceErrorType err,
+                             void * context)
+{
+    if (err != kDNSServiceErr_NoError)
+    {
+        ChipLogError(Discovery, "Mdns: %s Registered Record failed. Error: %d", __func__, err);
+        auto sdCtx = reinterpret_cast<RegisterRecordContext *>(context);
+        sdCtx->Finalize(err);
+    }
+};
+
 static void OnRegister(DNSServiceRef sdRef, DNSServiceFlags flags, DNSServiceErrorType err, const char * name, const char * type,
                        const char * domain, void * context)
 {
-    ChipLogDetail(Discovery, "Mdns: %s name: %s, type: %s, domain: %s, flags: %d", __func__, name, type, domain, flags);
-
+    ChipLogDetail(Discovery, "Mdns: %s name: %s, type: %s, domain: %s, flags: %d Error %d", __func__, name, type, domain, flags,
+                  err);
```

Comment:
`%s` and `Error::ToString(err)` ?

Reply:
Also is it really useful ? I think `RegisterContext::DispatchFailure` already logs if there is an error ?

Reply:
agreed will remove

Reply:
done